craft_cli/printer.py

```diff
@@ -416,3 +422,12 @@ def stop(self) -> None:
 
         self.log.close()
         self.stopped = True
+
+    def set_secrets(self, secrets: list[str]) -> None:
+        """Set the list of strings that should be masked out in all outputs."""
+        self.secrets = secrets[:]
```

Comment:
what's the objective of the `[:]`? do we need a method instead of just setting the attrib?

Reply:
This is to avoid client code modifying the list by accident after the fact, tested [here](https://github.com/canonical/craft-cli/pull/185/files#diff-026c46a62cfc6af61d84f79ea88769a17b59fb54e31346f67fdcc936918afd16R1154)
I suppose `secrets.copy()` would've been clearer, guess old habits die hard

Reply:
updated

Reply:
I would have though this would have only been a problem if the method had a default set to a list (where the common practice was to set to None to avoid this)